core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -25,28 +25,45 @@ case class SatoshisPerByte(currencyUnit: CurrencyUnit) extends BitcoinFeeUnit {
   def toSatPerKb: SatoshisPerKiloByte = {
     SatoshisPerKiloByte(currencyUnit.satoshis * Satoshis(1000))
   }
+
+  override def calc(tx: Transaction): CurrencyUnit =
+    Satoshis(tx.byteSize * toLong)
```

Comment:
Is this correct behavior for if `tx` is a `WitnessTransaction`? It might be but could you double check?

Reply:
It is. I am trying to find another library with multiple types for their fee rates but can't find one to prove it you though

Reply:
There is a test that tests the expected behavior for witness txs have different fees for sats/byte vs sats/vbyte